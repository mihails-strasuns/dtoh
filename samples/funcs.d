module funcs;

extern(C):

void foo_simple ();
void foo_with_body () { }
void foo_anon_params (int, double, char);
alias foo_t = long function(void*);
void hofoo1(foo_t);
void hofoo2(int function() param);
