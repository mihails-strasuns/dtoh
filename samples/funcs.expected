#include <stdint.h>

// Used enum definitions:



// Struct forward declaration to resolve cycles:



// Function pointer types:

typedef int64_t (*foo_t)(void*);

// Used struct definitions:



// Variable and function declarations:

void foo_simple();
void foo_with_body();
void foo_anon_params(int32_t, double, char);
void hofoo1(foo_t);
void hofoo2(int32_t (*param)());
